genomic_data_service/region_indexer_task.py

```diff
@@ -345,20 +355,21 @@ def metadata_doc(file_uuid, file_metadata, dataset_metadata):
             '@id': dataset_metadata['@id'],
             'target': list_targets(dataset_metadata),
             'target_label': get_target_label(dataset_metadata),
+            'disease_term_name': get_disease_term_name(dataset_metadata),
             'biosample_ontology': dataset_metadata.get('biosample_ontology', {}),
             'biosample_term_name': dataset_metadata.get('biosample_ontology', {}).get('term_name'),
             'documents': []
         },
         'dataset_type': dataset_metadata['@type'][0]
     }
-
-    for prop in REGULOME_COLLECTION_TYPES:
-        prop_value = dataset_metadata.get(prop)
-        assay_title = dataset_metadata.get('assay_title')
-        if assay_title == 'Histone ChIP-seq':
-            meta_doc['dataset']['collection_type'] = assay_title
-        elif prop_value:
-            meta_doc['dataset']['collection_type'] = prop_value
+    assay_title = dataset_metadata.get('assay_title')
+    if assay_title == 'Histone ChIP-seq':
+        meta_doc['dataset']['collection_type'] = assay_title
+    else:
+        for prop in REGULOME_COLLECTION_TYPES:
+            prop_value = dataset_metadata.get(prop)
+            if prop_value:
+                meta_doc['dataset']['collection_type'] = prop_value
```

Comment:
This is a little fragile and not future proof.  Theoretically, an Annotation can have both an annotation_type and an assay_term_name.    But it's only for a very specific thing (imputations or models of experiments)